File: src/semantic-router/pkg/logo/logo.go
package logo

import (
	"fmt"
)

// ANSI color codes
const (
	colorReset = "\033[0m"
	colorWhite = "\033[97m"
	colorMuted = "\033[38;2;145;158;171m"
)

func buildVLLMLogoLines() []string {
	return []string{
		"",
		colorWhite + `       █     █     █▄   ▄█` + colorReset,
		colorWhite + ` ▄▄ ▄█ █     █     █ ▀▄▀ █` + colorReset,
		colorWhite + `  █▄█▀ █     █     █     █` + colorReset,
		colorWhite + `   ▀▀  ▀▀▀▀▀ ▀▀▀▀▀ ▀     ▀` + colorReset,
		colorWhite + `  Semantic Router` + colorReset,
		colorMuted + `  Intelligent Routing for Mixture-of-Models` + colorReset,
		"",
	}
}

// PrintVLLMLogo prints the vLLM Semantic Router serve banner.
func PrintVLLMLogo() {
	for _, line := range buildVLLMLogoLines() {
		fmt.Println(line)
	}
}

File: src/vllm-sr/cli/logo.py
"""vLLM logo printing utilities."""

from __future__ import annotations

from cli.terminal import brand


def build_vllm_logo_lines() -> list[str]:
    """Build the serve banner lines using the installer wordmark style."""

    return [
        "",
        "       █     █     █▄   ▄█",
        " ▄▄ ▄█ █     █     █ ▀▄▀ █",
        "  █▄█▀ █     █     █     █",
        "   ▀▀  ▀▀▀▀▀ ▀▀▀▀▀ ▀     ▀",
        "  Semantic Router",
        "  Intelligent Routing for Mixture-of-Models",
        "",
    ]


def print_vllm_logo() -> None:
    """Print the vLLM Semantic Router serve banner."""
    brand("\n".join(build_vllm_logo_lines()))
